client/my-sites/plugins/plugin-install-button/index.jsx

```diff
@@ -261,12 +297,16 @@ export class PluginInstallButton extends Component {
 	}
 
 	renderNoticeOrButton() {
-		const { selectedSite, siteIsConnected } = this.props;
+		const { isWpcomAtomicSite, plugin, selectedSite, siteIsConnected } = this.props;
 
 		if ( siteIsConnected === false ) {
 			return this.renderUnreachableNotice();
 		}
 
+		if ( includes( IncompatiblePlugins, plugin.slug ) && isWpcomAtomicSite ) {
```

Comment:
Assuming that we will provide an `isCompatiblePlugin()` function, this doesn't matter, but in general, we don't need lodash in this case. We can just use the native [Array#includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) function like `IncompatiblePlugins.includes( plugin.slug )`.

Reply:
Hi! I've tried turning it into a Set as you suggested, however I've been struggling to get the function to work, being unable to check the plugin slug. When trying to use the includes as suggested (`incompatiblePlugins.includes( this.props.plugin.slug )`), I'm getting this error:

<img width="1145" alt="Screenshot 2020-02-03 at 17 58 38" src="https://user-images.githubusercontent.com/43215253/73677929-332c8c00-46af-11ea-8ac1-bd95cc480f30.png">

I wonder if you could please advise where I'm going wrong? :)

Reply:
Hi @Aurorum,

I'm sorry. I see how my comment could be confusing. This comment doesn't matter if you follow the advice in my [previous comment](https://github.com/Automattic/wp-calypso/pull/39020#discussion_r373077195) because you'll be calling an `isCompatiblePlugin()` function here instead of working directly with an Array (or Set).

My intent was simply to point out that the `Array#includes` method should probably be used instead of lodash's `includes()` function because it is a native part of the Array API and allows us to reduce our dependency on lodash. In general, if we can do without a dependency while writing similar code, we prefer to do without because dependencies increase the size of the JS bundles we send to the browser.

If you are in an intermediate place where you have converted `incompatiblePlugins` to a `Set` but are still exporting `incompatiblePlugins` and not an `isCompatiblePlugin()` function, the `includes()` method is not provided by `Set`. Instead, you can use the [Set#has()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has) method like `IncompatiblePlugins.has( plugin.slug )`.

Does that clear things up?

Reply:
Thank you, it does! Sorry, last question: I've been toying around a lot with trying to make the function you referenced work, and I've been searching through Calypso on a way to get the plugin slug. I've been unable to find anything though, so whilst everything else works, the main logic on `if the slug has something in the set` doesn't, since I'm not sure how to get the slug. Any ideas? :)

Reply:
Ah, this is again a place where I could have been clearer. The `isCompatiblePlugin` function should take a slug argument like `isCompatiblePlugin( pluginSlug )`. When I mentioned `isCompatiblePlugin()`, I meant the parentheses to indicate it is a function, but I didn't mean to indicate the function should take no arguments. I'm sorry for the confusion.

If you're asking where the React components in this PR get the plugin slug, it looks like it is part of the `plugin` prop they are receiving (i.e. `this.props.plugin.slug`).

Does that answer your question?

Reply:
Yes, thank you very much! I believe that this PR should now be working with the ideal function, and is ready for review again. :)